### PR TITLE
Test refactor

### LINKED ISSUE #45: Refactor Integration and acceptance tests
Use a docker-compose file to automatically run the integration and acceptance with just a simple gradle task.
Also add the testing using docker containers to travis for better CI

### PR DESCRIPTION
**Pull request by yatharthranjan**
**Closes #45 
Depends on #44** 

In this PR - 
- Refactored and improved integration and acceptance tests
- Use docker compose specific to tests
- just one gradle command to bring up the containers, run the tests and then stop the containers.
- Add the testing process to travis build for better CI

### CODE REVIEW

**Comment by afolarin** [comment]
@yatharthranjan I've blocked some time to review this today -- apologies for the wait!

**Comment by yatharthranjan** [comment]
No problems. Thank You.

File: README.md [hunk 1]
```diff
@@ -180,37 +172,18 @@ Although cogstack has unit tests where appropriate, the nature of the project is
...
-
-to launch all of the external services.
-
-
-All being well, you should now be able to run the integration tests. Each of these demonstrate a different facet of cogstack's functionality.
-Each integration test follows the same pattern:
-
+A single command can be used to run the integration tests. Each of these demonstrate a different facet of cogstack's functionality.
+Each time an integration test is run, it follows the same pattern:
+* Start up the required services using docker-compose
```
Comment (afolarin) [inline README.md:182]: should probably add a instruction on which directory to run docker-compose up from?

**Review by afolarin — APPROVED** [review]
LGTM

**Pull request merged**